### PR TITLE
Feat: add no-non-null-assertion rule

### PR DESCRIPTION
**Pull request by iykekings**

### CODE REVIEW

File: src/rules/no_non_null_assertion.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
+use super::Context;
```
Comment (disizali) [inline src/rules/no_non_null_assertion.rs:1]: copyright

File: src/rules/mod.rs [hunk 2]
```diff
@@ -33,6 +33,7 @@ mod no_func_assign;
 mod no_misused_new;
 mod no_namespace;
 mod no_new_symbol;
+mod no_non_null_assertion;
```
Comment (disizali) [inline src/rules/mod.rs:36]: can you add this rule to the Readme too, please?

File: src/rules/no_non_null_assertion.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    assert_lint_err::<NoNonNullAssertion>("x!!!;", 0);
+    assert_lint_err::<NoNonNullAssertion>("x.y?.z!();", 0);
+    assert_lint_err::<NoNonNullAssertion>("x.y.z!();", 0);
+    assert_lint_err::<NoNonNullAssertion>("x![y]?.z;", 0);
+    assert_lint_err::<NoNonNullAssertion>("x![y];", 0);
+    assert_lint_err::<NoNonNullAssertion>("!x!.y;", 1);
+    assert_lint_err::<NoNonNullAssertion>("x!.y?.z;", 0);
+    assert_lint_err::<NoNonNullAssertion>("x.y!;", 0);
+    assert_lint_err::<NoNonNullAssertion>("x!.y;", 0);
+    assert_lint_err::<NoNonNullAssertion>("x!;", 0);
```
Comment (disizali) [inline src/rules/no_non_null_assertion.rs:81]: nice tests

File: src/rules/mod.rs (hunk 2 above)
Comment (iykekings) [reply src/rules/mod.rs:36]: Yes I have. Thanks

**Pull request merged**